### DISCUSSION TITLE
Build a AIO client/server

### DISCUSSION THREAD

**Discussion by webysther** [uniuuu/zotprime#23] [Ideas]
Hi, any chance to create a all-in-one server using supervised?

**Comment by uniuuu** [comment]
Hi @webysther 
Could you please share an URL of the supervised?

**Comment by uniuuu** [comment]
> > @webysther It's a game changer. Bravo! I see it's deprecated in favor of https://docs.linuxserver.io/images/docker-zotero/#supported-architectures
> 
> Yeap, they created! The guys of the project are incredable! I just think in a good way to create a some sort of AIO of all services of zotero only for the server and want to discuss with you, what you think? You make a really organized job here


> Maybe a AIO like nextcloud AIO docker or one container only for stream-server and dataserver and tinymce-clean-server to reduce the amount of containers.


@webysther 
It's possible to put all components into one container but it is an effort. What is your use case why would you need one big container for all?
Initially in the latest project the focus was exactly opposite to split it all into a separate services and run it in kubernetes. 
This way it's easier to port it to a cloud, manage, scale, monitor.

**Reply by webysther** [reply]
For small self hosted homelab like me, maybe I move to k8s in future but for my 2 user base don't make any sense. To use AIO image is fast and used more easy by community, a good example I start using is immich with this image: https://github.com/imagegenius/docker-immich

**Reply by uniuuu** [reply]
@webysther Have you looked into [MicroK8s](https://github.com/canonical/microk8s) as it's small and fit homelab?
Even you can use LXD virtualisation to run MicroK8s in its container! [MicroK8s in LXD]( https://microk8s.io/docs/install-lxd)

> Maybe a AIO like nextcloud AIO docker or one container only for stream-server and dataserver and tinymce-clean-server to reduce the amount of containers.

It's much easier to manage updates/deprecation of components/libs when components are split. Originally both tinymce-clean-server and stream-server were run under dataserver container refer to https://github.com/FiligranHQ/zotprime/blob/master/docker/entrypoint.sh#L22 . 

But because during update they've got errors when trying to switch to more latest Node.js therefore moved them out.

The tinymce-clean-server never updated https://github.com/zotero/tinymce-clean-server, plan to remove it once confirmed it's no longer.

**Comment by uniuuu** [comment]
Link to original posts:
https://github.com/uniuuu/zotprime/issues/20
https://github.com/uniuuu/zotprime/issues/19

**Comment by uniuuu** [comment]
> For small self hosted homelab like me, maybe I move to k8s in future but for my 2 user base don't make any sense. To use AIO image is fast and used more easy by community, a good example I start using is immich with this image: https://github.com/imagegenius/docker-immich

Hi @webysther 
This project has moved forward with separated containers and microk8s delivers remarkable results. You have all in one, HA if you need, certificate management with ingress, self-healing, built-in monitoring, resources management just to name few and what's most important it runs in a **homelab**. 

> MicroK8s runs in as little as 540MB of memory, but to accommodate workloads, we recommend a system with at least 20G of disk space and 4G of memory.

All-in-one container will be harder to manage as it's more fragile since the single point of failure. 

Please try latest code you can run it in ARM64 too now.
